drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -1144,6 +1144,7 @@ void nrf_802154_energy_detection_failed(nrf_802154_ed_error_t error)
 void nrf_802154_serialization_error(const nrf_802154_ser_err_data_t *err)
 {
 	__ASSERT(false, "802.15.4 serialization error: %d", err->reason);
+	k_panic();
```

Comment:
Actually, as both of this files are in zephyr we could consider wrapping up `k_panic` in `#ifndef CONFIG_ASSERT`, as the zephyr assert will call `k_panic()`/`k_oops` anyway.

Reply:
I'm against. Did you know about `CONFIG_FORCE_NO_ASSERT` or `CONFIG_ASSERT_LEVEL`? This controls internal macro __ASSERT_ON in [zephyr/include/zephyr/sys/__assert.h](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/__assert.h). Who knows if somebody adds some hypothetical additional dependencies that control behavior of asserts like `CONFIG_ASSERT_BUT_DONT_HALT` or `CONFIG_ASSERT_EVEN_IF_FORCE_NO_ASSERT_IS_ENABLED`, our assuption that nrf_802154_serialization_error does not return would be broken again. You know what I mean? 

I would like to have clear code without any `if`s, the code that will guarantee the `nrf_802154_serialization_error` does not return. I don't want to read zephyr/.config logs to check if this function could return is some circumstances